libp2p/switch.nim

```diff
@@ -253,16 +253,14 @@ proc start*(s: Switch): Future[seq[Future[void]]] {.async, gcsafe.} =
 
   for s in startFuts:
     if s.failed:
-      info "Failed to start one transport", error=s.error.msg
+      error "Failed to start one transport", error=s.error.msg
```

Comment:
So we would throw here (and in the future, add the callback)

(Or better, get rid of startFuts and simplify the logic to a single loop)

Reply:
I think the multiple loops here are warranted as we build up a sequence of Futures and let them all complete asynchronously. A single loop would require us to await the start in each iteration.

There is now an exception raised here, with a TODO note about replacing with a `listenError` callback.

Reply:
Yeah, in practice, launching a transport is nearly instantaneous (as proof, before a few month ago, we weren't actually waiting for them and everything was working most of the time, see https://github.com/status-im/nim-libp2p/pull/609)
so launching them in parallel is a bit useless in terms of speed (at least with current transports, but I guess with future ones too), however it makes the logic more complex

But we can also refacto that as part of #667